### PR TITLE
Mappings

### PR DESCRIPTION
**Pull request by ncordon**
Adds generation of a mappings file of shape:
```
RNAID¹†TaxID¹,TaxID², TaxID³,...
RNAID²†TaxID'¹,TaxID'², TaxID'³,...
...
```

Closes #98 and #99

### CODE REVIEW

File: notes/4.0.0.markdown [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Each database is now composed of a pair of files: 
+* `rna16s.fa`: stores the sequences with their `RNAID`s.
+* `mappings`: stores for each `RNAID` a row of shape `RNAID¹†TaxID¹,TaxID², TaxID³,...`, where we have a list of associated taxon IDs for a given `RNAID`.
+
+It is guaranteed that for each `RNAID` in the `rna16s.fa` file, there is an associated entry in the mappings file.
+
+To read a mappings file, namely `mappings` we have added an `IO` module and it suffices to do:
+
+```
+import ohnosequences.files.read
```
Comment (agarciamontoro) [inline notes/4.0.0.markdown:21]:
```suggestion
import ohnosequences.files.read
import ohnosequences.db.rna16s.io.deserializeMappings
```

File: src/main/scala/io.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        // All upper case, U -> T
+        Sequence(entry.rnaSequence.sequence.toUpperCase.replace('U', 'T'))
+    )
+
+  /** Recieves an Iterator over the lines of a file, and reads them into
+    * a [[Mappings]] structure RNAID -> [TaxID]
+    */
+  def deserializeMappings(lines: Lines): Mappings =
+    lines.map { line =>
+      val values = line.split("†")
```
Comment (agarciamontoro) [inline src/main/scala/io.scala:29]: It's ok for now, I would not change it, but maybe we could add a `serialOptions` case class that contain the separators we expect.

File: src/main/scala/io.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      val rnaID = values(0)
+      val taxIDs = values(1).split(",").toSet.map { s: String =>
+        s.toInt
+      }
+
+      (rnaID, taxIDs)
+    }.toMap
+
+  /** Recieves a [[Mappings]] structure and produces a serialized
```
Comment (agarciamontoro) [inline src/main/scala/io.scala]: Typo!

```suggestion
  /** Receives a [[Mappings]] structure and produces a serialized 
```

**Review by agarciamontoro — APPROVED** [review]
LGTM!

Minor comments below.

File: src/main/scala/io.scala (hunk 2 above)
Comment (ncordon) [reply src/main/scala/io.scala:29]: I agree

**Pull request merged**